src/Estimators/include/BipedalLocomotion/RobotDynamicsEstimator/SubModelKinDynWrapper.h

```diff
@@ -0,0 +1,199 @@
+     */
+    bool updateDynamicsVariableState();
+
+public:
+    /**
+     * @brief initialize initializes the kindyncomputation object and resizes
+     * all the variable not allocated yet.
+     * @param subModel a struct of type SubModif el describing the subModel structure and model.
+     * @param kinDynFullModel is the KinDynComputation object associated to a model.
+     * @return a boolean value saying if all the variables are initialized correctly.
+     */
+    bool initialize(SubModel & subModel,
+                    std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel);
+
+    /**
+     * @brief updateKinDynState updates the state of the KinDynWrapper object.
+     * @param kinDynFullModel is the KinDynComputation object associated to a model.
+     * @return a boolean value saying if the subModelList has been created correctly.
+     */
+    bool updateKinDynState(std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel);
+
+    /**
+     * @brief inverseDynamics computes the free floaing inverse dynamics
+     * @param motorTorqueAfterGearbox a vector of size number of joints containing the motor torques
+     * times the gearbox ratio.
+     * @param frictionTorques a vector of size number of joints containing the friction torques.
+     * @param tauExt a vector of size number of joints containing the joint torques generated by the
+     * external contacts.
+     * @param baseAcceleration the sub-model base acceleration.
+     * @return a boolean value
+     */
+    bool inverseDynamics(Eigen::VectorXd& motorTorqueAfterGearbox,
+                         Eigen::VectorXd& frictionTorques,
+                         Eigen::VectorXd& tauExt,
+                         Eigen::VectorXd& baseAcceleration,
+                         Eigen::VectorXd& jointAcceleration);
+
+    /**
+     * @brief getBaseAcceleration gets the acceleration of the sub-model base
+     * @param kinDynFullModel is the KinDynComputation object associated to a model.
+     * @param robotBaseAcceleration the acceleration of the robot base.
+     * @param robotJointAcceleration the acceleration of the robot joints.
+     * @param subModelBaseAcceleration the base acceleration of the sub-model.
+     * @return a boolean value.
+     */
+    bool getBaseAcceleration(std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel,
+                             const Eigen::VectorXd& robotBaseAcceleration,
+                             const Eigen::VectorXd& robotJointAcceleration,
+                             Eigen::Ref<Eigen::VectorXd> subModelBaseAcceleration);
+
+    /**
+     * @brief getBaseVelocity gets the acceleration of the sub-model base
+     * @param kinDynFullModel is the KinDynComputation object associated to a model.
+     * @return baseVelocity the the base velocity of the sub-model.
+     */
+    Eigen::Ref<Eigen::VectorXd> getBaseVelocity(std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel);
```

Comment:
This gives the user the possibility to change the base velocity is this wanted? 
```suggestion
    Eigen::Ref<const Eigen::VectorXd> getBaseVelocity(std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel) const;
```

Reply:
Actually this should be a `manif::SE3d::Tangent` object